pallets/file-system/src/lib.rs

```diff
@@ -313,11 +400,16 @@ pub mod pallet {
             let start_block = NextStartingBlockToCleanUp::<T>::get();
             let mut block_to_clean = start_block;
 
-            // Total weight used to avoid exceeding the remaining weight
+            let writes = match T::MaxExpiredStorageRequests::get().checked_add(1) {
+                Some(writes) => writes,
+                None => return Weight::zero(),
```

Comment:
```suggestion
                // This should never happen. It would mean that MaxExpiredStorageRequests is u32::MAX,
                // which is an irrational number to set as a limit.
                None => return Weight::zero(),
```